### ISSUE TITLE
[Messenger] Ability to dispatch retryEnvelope to another bus

### ISSUE THREAD

**Issue by proArtex** [symfony/symfony#31848]
I have separate buses like this:
```yaml
# messenger.yaml

framework:
    messenger:
        default_bus: my.sender
        buses:
            my.sender:
                default_middleware: false
                middleware:
                    - messenger.middleware.send_message
            my.receiver:
                default_middleware: false
                middleware:
                    - App\Middleware\MyMiddleware
        transports:
            async:
                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
                retry_strategy:
                    max_retries: 999
                    delay: 10000
                    multiplier: 1
                    max_delay: 0
        routing:
             '*': async
```

I use command `bin/console messenger:consume async --bus=my.receiver`.

So  we have  a retryEnvelope dispatch call that goes directly to the same receiver bus. Variable $bus comes from ConsumeMessagesCommand:
```php
# ConsumeMessagesCommand.php
$bus = $this->busLocator->get($input->getOption('bus'));

# Worker.php
$this->bus->dispatch($retryEnvelope);
```

Since I have different buses (I strongly believe that one bus per send/receive ops is kind of workaround with "ifs"), I need an ability to somehow point to another bus. Is there any solution coming? Did I miss something?

**Comment by Tobion** [comment]
I was planning to extract the retry logic in https://github.com/symfony/symfony/blob/926ded8ea5017cb8aaba834f418d1c8d8337dabc/src/Symfony/Component/Messenger/Worker.php#L149-L155 into a listener for WorkerMessageFailedEvent. Then you could overwrite the listener to dispatch retries to a different bus if you want.

**Comment by Tobion** [comment]
@proArtex What I don't understand is why you have different buses for send and receive? I don't see the potential use-case for this. What are you trying to solve?

**Comment by proArtex** [comment]
@Tobion first I wanted to dispatch a new message inside consumer, that became possible with the last update (dispatch_after_current_bus middleware). But I still have absolutely different middleware stack for sending and receiving messages. In case of one bus I have to use "if" in every middleware to check whether it is sending or receiving. Imo it's better to put "ifs" into configuration of different buses.

**Comment by Tobion** [comment]
Ok makes sense to me. This should be possible once https://github.com/symfony/symfony/issues/31848#issuecomment-498705251 is implemented. You could then overwrite/change the listener and use a different bus.
@proArtex would you want to work on the proposal? Basically extract the hardcoded try-logic inside the Worker into event listeners. The events are already there.

**Comment by proArtex** [comment]
@Tobion have absolutely no time for now, sorry.

**Comment by Tobion** [comment]
Fix in #34185 that allows to you to inject a different sender service locator in SendFailedMessageForRetryListener that returns the my.sender for my.receiver
